nameko/testing/utils.py

```diff
@@ -140,3 +140,65 @@ def find_free_port(host='127.0.0.1'):
     port = sock.getsockname()[1]
     sock.close()
     return port
+
+
+class ResourcePipeline(object):
+    """ Creates and destroys resources in background threads.
+
+    Creates up to `size` resources ahead of time so the caller avoids waiting
+    for lazy creation.
+    """
+
+    STOP = object()
+
+    def __init__(self, create, destroy, size=3):
+        if size == 0:
```

Comment:
`size=3`? From trial and error in the vhost usage -- 3 resulted in the best speedup. It's a balance between having something available when you need it and the "waste" of destroying what's left unused at the end of the run.